sentinel-adapter/sentinel-apache-dubbo-adapter/src/main/java/com/alibaba/csp/sentinel/adapter/dubbo/SentinelDubboConsumerFilter.java

```diff
@@ -53,33 +54,30 @@ public SentinelDubboConsumerFilter() {
     public Result invoke(Invoker<?> invoker, Invocation invocation) throws RpcException {
         Entry interfaceEntry = null;
         Entry methodEntry = null;
+        RpcContext rpcContext = RpcContext.getContext();
         try {
-            String resourceName = DubboUtils.getResourceName(invoker, invocation, DubboConfig.getDubboConsumerPrefix());
-            interfaceEntry = SphU.entry(invoker.getInterface().getName(),
-                ResourceTypeConstants.COMMON_RPC, EntryType.OUT);
-            methodEntry = SphU.entry(resourceName, ResourceTypeConstants.COMMON_RPC, EntryType.OUT);
+            String methodResourceName = DubboUtils.getResourceName(invoker, invocation, DubboConfig.getDubboConsumerPrefix());
+            String interfaceResourceName = DubboConfig.getDubboInterfaceGroupAndVersionEnabled() ? invoker.getUrl().getColonSeparatedKey()
+                    : invoker.getInterface().getName();
+            InvokeMode invokeMode = RpcUtils.getInvokeMode(invoker.getUrl(), invocation);
 
-            Result result = invoker.invoke(invocation);
-            if (result.hasException()) {
-                Throwable e = result.getException();
-                // Record common exception.
-                Tracer.traceEntry(e, interfaceEntry);
-                Tracer.traceEntry(e, methodEntry);
+            if (InvokeMode.SYNC == invokeMode) {
+                interfaceEntry = SphU.entry(interfaceResourceName, ResourceTypeConstants.COMMON_RPC, EntryType.OUT);
+                methodEntry = SphU.entry(methodResourceName, ResourceTypeConstants.COMMON_RPC, EntryType.OUT, invocation.getArguments());
+            } else {
+                // should generate the AsyncEntry when the invoke model in future or async
+                interfaceEntry = SphU.asyncEntry(interfaceResourceName, ResourceTypeConstants.COMMON_RPC, EntryType.OUT);
+                methodEntry = SphU.asyncEntry(methodResourceName, ResourceTypeConstants.COMMON_RPC, EntryType.OUT, 1, invocation.getArguments());
             }
-            return result;
+            rpcContext.set(DubboUtils.DUBBO_INTERFACE_ENTRY_KEY, interfaceEntry);
+            rpcContext.set(DubboUtils.DUBBO_METHOD_ENTRY_KEY, methodEntry);
+            return invoker.invoke(invocation);
         } catch (BlockException e) {
             return DubboFallbackRegistry.getConsumerFallback().handle(invoker, invocation, e);
         } catch (RpcException e) {
-            Tracer.traceEntry(e, interfaceEntry);
-            Tracer.traceEntry(e, methodEntry);
             throw e;
```

Comment:
Missing `traceAndExit()` here?

Reply:
No. It's also unnecessary to catch the RpcException in `SentinelDubboProvider`  and `SentinelDubboConsumer` to trace. Dubbo will catch the Exception when exec `filter.invoke` and call the `onError` to handle this.